NavigationService/Navigation.cs

```diff
@@ -375,6 +305,130 @@ public string GetNearestRoute()
+                        {
+                            nearestList.Add(station.distancefromstar ?? 0, station.name);
+                        }
+                    }
+
+                    // Station is nearest to the main star which meets the service query
+                    ServiceStation = nearestList.Values.FirstOrDefault();
+
+                    // Get mission IDs for 'service' system
+                    missionids = ((MissionMonitor)EDDI.Instance.ObtainMonitor("Mission monitor"))?.GetSystemMissionIds(ServiceSystem);
+
+                    // Set route and destination variables
+                    missionMonitor.SetNavigationData(ServiceSystem, ServiceStation, ServiceDistance);
+                }
+            }
+            EDDI.Instance.enqueueEvent(new RouteDetailsEvent(DateTime.Now, serviceType, ServiceSystem, ServiceSystem, missionids.Count(), ServiceDistance, ServiceDistance, missionids));
+            return ServiceSystem;
+        }
+
+        public StarSystem GetServiceSystem(string serviceType, int maxStationDistance, bool prioritizeOrbitalStations)
+        {
+            StarSystem currentSystem = EDDI.Instance?.CurrentStarSystem;
+            if (currentSystem != null)
+            {
+                // Get the filter parameters
+                string shipSize = EDDI.Instance?.CurrentShip?.size ?? "Large";
+                ServiceFilter.TryGetValue(serviceType, out dynamic filter);
+                int cubeLy = filter.cubeLy;
+
+                //
+                List<string> checkedSystems = new List<string>();
+                string ServiceSystem = null;
+                int maxTries = 5;
+
+                while (ServiceSystem == null && maxTries > 0)
+                {
+                    List<StarSystem> cubeSystems = StarMapService.GetStarMapSystemsCube(currentSystem.name, cubeLy);
+                    if (cubeSystems?.Any() ?? false)
+                    {
+                        // Filter systems using search parameters
+                        cubeSystems = cubeSystems.Where(s => s.population >= filter.population).ToList();
+                        cubeSystems = cubeSystems.Where(s => filter.security.Contains(s.security)).ToList();
+                        if (serviceType != "facilitator")
+                        {
+                            cubeSystems = cubeSystems
+                                .Where(s => filter.econ.Contains(s.Economies.FirstOrDefault(e => e.invariantName != "None")?.invariantName))
+                                .ToList();
+                        }
+
+                        // Retreive systems in current radius which have not been previously checked
+                        List<string> systemNames = cubeSystems.Select(s => s.name).Except(checkedSystems).ToList();
+                        List<StarSystem> StarSystems = StarSystemSqLiteRepository.Instance.GetOrFetchStarSystems(systemNames.ToArray(), true, false);
```

Comment:
Since we are fetching systems but not creating / saving star systems, we can afford to retrieve less complete results.
Let's make the `GetOrFetchStarSystems()` method more efficient for this query. Proposed revision as follows:
```
        public List<StarSystem> GetOrFetchStarSystems(string[] names, bool fetchIfMissing = true, bool refreshIfOutdated = true, bool showCoordinates = true, bool showSystemInformation = true, bool showBodies = true, bool showStations = true, bool showFactions = true)
        {
            if (names.Count() == 0) { return null; }

            List<StarSystem> systems = Instance.GetStarSystems(names, refreshIfOutdated);
            List<string> fetchSystems = new List<string>();

            // If a system isn't found after we've read our local database, we need to fetch it.
            foreach (string name in names)
            {
                if (fetchIfMissing && systems.FirstOrDefault(s => s.name == name) == null)
                {
                    fetchSystems.Add(name);
                }
            }

            List<StarSystem> fetchedSystems = DataProviderService.GetSystemsData(fetchSystems.ToArray(), showCoordinates, showSystemInformation, showBodies, showStations, showFactions);
            if (fetchedSystems?.Count > 0)
            {
                Instance.insertStarSystems(fetchedSystems);
                systems.AddRange(fetchedSystems);
            }

            return systems;
        }
```

Reply:
You may want to review the difference between 'GetOrFetch' and 'GetOrCreate'.

Both save the the system to the database, but only 'Create' will make a placeholder if the system is not found.

Reply:
Ok. Then let's amend my proposal to add an optional bool for saving the results.

Reply:
Personally, I think adding those options is overkill. You do not save significant time on an EDSM call, just pulling station data... I tested this.

Additionally, not saving the `fetchIfMissing` systems to the database creates more future EDSM calls for systems you've previously touched on a past station service query.

My strategy was to be a good EDSM citizen and only pull detailed data on a system once.